webapp/utils/aws/cloudwatch.py

```diff
@@ -51,38 +51,17 @@ def get_log_events(self, task_id: str) -> list:
             "startFromHead": True,
         }
 
-        if self.log_stream_exists(task_id):
-            while True:
-                response = self.client.get_log_events(**params)  # type: ignore[arg-type]
-                log_events.extend(response["events"])
-                next_token = response.get("nextForwardToken")
-                if next_token == params.get("nextToken"):
-                    # the end of the stream is marked by returning the same token
-                    break
-                params["nextToken"] = next_token
-            logger.info("CloudWatch logs retrieved.")
-            return log_events
-        raise ECSTaskLogStreamDoesNotExistError(task_id)
-
-    def log_stream_exists(self, task_id: str) -> bool:
-        return any(
-            task_id == log_stream_name.split("/")[-1]
-            for log_stream_name in self.get_log_streams()
-        )
-
-    def get_log_streams(self) -> list[str]:
-        log_streams: list[str] = []
-        params = {
-            "logGroupName": self.log_group_name,
-        }
         while True:
-            response = self.client.describe_log_streams(**params)  # type: ignore[arg-type]
-            log_streams.extend(
-                logstream["logStreamName"] for logstream in response["logStreams"]
-            )
-            next_token = response.get("nextToken")
-            if next_token is None:
-                # all log streams retrieved is "nextToken" excluded from response
```

Comment:
Cutting is delivering!  ❤️ 

In all seriousness, nice work here.  It's hard to cut code, especially new code that you just wrote.  But to my eyes, feels like you've found a more direct and straight-forward approach here.

Reply:
Yes, this feels more streamlined!